client/src/document/index.js

```diff
@@ -41,12 +44,20 @@ export function Document(props) {
     setLoading(false);
   }, [loadingError]);
 
-  async function fetchDocument(pathname) {
-    let url = pathname;
-    if (!url.endsWith("/")) {
-      url += "/";
+  const getCurrentPathname = useCallback(() => {
```

Comment:
Can't we use
https://github.com/ReactTraining/react-router/blob/f65d9b951435da57fff561f7da0f6a759fb36c18/packages/react-router/index.js#L441
for that?

Reply:
First of all; as mentioned in Slack, I don't even know how to use this. 

Second of all, now I remember why I'm doing it this way. First of all, I want it to be a function I can control because when you're using the `server` (`localhost:5000`) the Express will force the URLs to end with a `/` so I want to have control over that. 

Secondly, when I worked on the "edit-in-web-app" side-project, I used `/en-US/edit/HTML` to load a different wrapper. So that `<DocumentEdit/>` actually renders `<Document/>` within. So if I use `useLocation()`  inside the `<Document/>` component it's going to use the location that was used to load `<DocumentEdit/>` which I'm not interested in. 

What will really help me explain my reasoning is if I give this function a better name! Instead of `getCurrentPathname` I think `getCurrentMdnUrl` (or `getCurrentDocumentUri`) might be better. 
But there might be an even better opportunity! So the WebSocket will report when a document has been updated, and I need to know if the document that changed was the one I'm currently have in the browser tab. But if that could be rewritten to be like this:

```javascript
  function onMessage(data) {
    if (data.slug === params['*'] && data.locale === params.locale) {
      // The recently edited document is the one we're currently looking at!
      fetchDocument();
    }
  }
```
...then we might be better equipped to have the knowledge of whether we should call `fetchDocument` INDEPENDENT OF URLs and purely based on `locale` & `slug` combos. 
Hold my beer.

Reply:
I like it! 🍺✋

Reply:
Nevermind. The watcher builds documents. Meaning, it builds each `index.json`. That JSON does not have `.locale` and `.slug`. The reason is probably because it's not needed. It does have a `.mdn_url` which I'm starting to hate but to change that here and now might be too big of an idea.

Reply:
There's something something going on, that doesn't feel right, about the mix between `locale` & `slug` as a combo vs. them being serializing in a URL. For example, the Builder is very much about URLs but the react-router stuff is very much about params. And then there's the watcher and the server. All different pieces trying to communicate with each other. I'm pretty sure a serialized URL wins in practicality and versatility when it comes to communication via HTTP GET which is what we're doing between the React dev server and the `server` (`localhost:5000`). 

But we're probably not done. We might need to change our minds again some time in the future. 

PS. I did rename it to `getCurrentDocumentUri` since the word `pathname` feels too much exclusively related to `window.location` or `useLocation().pathname`.